pyinfra/facts/iptables.py

```diff
@@ -0,0 +1,74 @@
+# pyinfra
+# File: pyinfra/facts/iptables.py
+# Desc: facts for the Linux iptables firewall
+
+from __future__ import unicode_literals
+from collections import OrderedDict
+
+from pyinfra.api import FactBase
+
+IPTABLES_ARGS = {
+    '--dport': 'dport',
+    '-p': 'protocol',
+    '--to-destination': 'destination',
+    '-i': 'interface',
+}
+
+
+def parse_rule(line):
+    "Parse one iptables rule"
+    line = line.split()
+    keys = line[::2]
+    values = line[1::2]
+    # tuples with raw iptable argument names
+    raw_definition = zip(keys, values)
+    # dict with keys mapped to readable names when known
+    return {
+        IPTABLES_ARGS.get(key, key): value
+        for key, value in raw_definition
+    }
+
+
+def parse_rules(output):
+    "Parse iptables rules from an output list"
+    for line in output:
+        if line.startswith('-'):
+            yield parse_rule(line)
+
+
+class Iptables(FactBase):
+    '''
+    Sorted and parsed list of iptables rules from a specific table
+    '''
+    def command(self, name='nat'):
+        return 'iptables-save -t {0}'.format(name)
+
+    def process(self, output):
+        return list(parse_rules(output))
+
+
+class IptablesForward(Iptables):
```

Comment:
I don't think this fact is needed - the `Iptables` fact should provide everything that's needed - and the module should do the actual state comparison using `host.fact.iptables`.